osm2lanes-web/src/lib.rs

```diff
@@ -363,3 +376,34 @@ impl App {
         }
     }
 }
+
+fn generate_test_yaml(road: &Road, tags: &str, locale: &Locale, way_id: Option<i64>) -> String {
+    use osm2lanes::test::{Expected, TestCase};
+
+    let test = TestCase {
+        way_id,
+        link: None,
+        comment: None,
+        description: Some("fill me out".to_string()),
+        example: None,
+        driving_side: locale.driving_side,
+        iso_3166_2: locale.iso_3166_2_subdivision.clone(),
+        tags: Tags::from_str(tags).unwrap(),
+        expected: Expected::Road(road.clone()),
+        rust: None,
+    };
+    // TODO Strip out road's name, ref, and other things we don't test for?
+    // TODO Based on checkboxes, strip out separators
+    let raw = serde_yaml::to_string(&test).unwrap();
+
+    // serde_yaml explicitly lists "field: null" for None values. Filter these out, to match the
+    // style of the test YAML.
+    let mut output = String::new();
```

Comment:
The output isn't indented properly for copying into the .yaml, and I'm having trouble getting this working... will iterate later